src/components/Search/SearchPageHeader/SearchTypeMenuButton.tsx

```diff
@@ -0,0 +1,200 @@
+import React, {useCallback, useMemo, useRef, useState} from 'react';
+import {View} from 'react-native';
+import type {TextStyle, ViewStyle} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import Button from '@components/Button';
+import type {MenuItemWithLink} from '@components/MenuItemList';
+import {usePersonalDetails} from '@components/OnyxProvider';
+import PopoverMenu from '@components/PopoverMenu';
+import type {PopoverMenuItem} from '@components/PopoverMenu';
+import type {SearchQueryJSON} from '@components/Search/types';
+import ThreeDotsMenu from '@components/ThreeDotsMenu';
+import useActiveCentralPaneRoute from '@hooks/useActiveCentralPaneRoute';
+import useDeleteSavedSearch from '@hooks/useDeleteSavedSearch';
+import useLocalize from '@hooks/useLocalize';
+import useSingleExecution from '@hooks/useSingleExecution';
+import useStyledSafeAreaInsets from '@hooks/useStyledSafeAreaInsets';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import {clearAllFilters} from '@libs/actions/Search';
+import {mergeCardListWithWorkspaceFeeds} from '@libs/CardUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import type {AuthScreensParamList} from '@libs/Navigation/types';
+import {getAllTaxRates} from '@libs/PolicyUtils';
+import {buildSearchQueryJSON, buildUserReadableQueryString, isCannedSearchQuery} from '@libs/SearchQueryUtils';
+import {createBaseSavedSearchMenuItem, createTypeMenuItems, getOverflowMenu as getOverflowMenuUtil} from '@libs/SearchUIUtils';
+import * as Expensicons from '@src/components/Icon/Expensicons';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {SaveSearchItem} from '@src/types/onyx/SaveSearch';
+
+type SavedSearchMenuItem = MenuItemWithLink & {
+    key: string;
+    hash: string;
+    query: string;
+    styles?: Array<ViewStyle | TextStyle>;
+};
+
+type SearchTypeMenuNarrowProps = {
+    queryJSON: SearchQueryJSON;
+};
+
+function SearchTypeMenuButton({queryJSON}: SearchTypeMenuNarrowProps) {
```

Comment:
```suggestion
function Search<xxx>MenuPopover({queryJSON}: SearchTypeMenuNarrowProps) {
```

Why?
 - the component renders BOTH button and Popover - so it's not obvious that Button is more important 
 - if we look at what this component actualy does - 80-90% of its code is related to displaying list of specific items in a popover; only small parts of it render the button
 - as for the "type" part - it's not only a list of different types right? we need a better name; Type is meaningless in this context (unless Im mistaken)